internal/server/singleprocess/service_job.go

```diff
@@ -81,28 +81,14 @@ func (s *service) QueueJob(
 	}
 
 	// Verify the project exists and use that to set the default data source
-	project, err := s.state.ProjectGet(&pb.Ref_Project{Project: job.Application.Project})
+	_, err := s.state.ProjectGet(&pb.Ref_Project{Project: job.Application.Project})
 	if status.Code(err) == codes.NotFound {
 		return nil, status.Errorf(codes.NotFound,
-			"Project %s was not found! Please ensure that 'waypoint init' was run with this project.",
+			"Project %q was not found! Please ensure that 'waypoint init' was run with this project.",
 			job.Application.Project,
 		)
 	}
 
-	if job.DataSource == nil {
-		if project.DataSource == nil {
-			return nil, status.Errorf(codes.FailedPrecondition,
-				"Project %s does not have a data source configured. Remote jobs "+
-					"require a data source such as Git to be configured with the project. "+
-					"Data sources can be configured via the CLI or UI. For help, see : "+
-					"https://www.waypointproject.io/docs/projects/git#configuring-the-project",
-				job.Application.Project,
```

Comment:
I understand why you got rid of this error. 

Did this error go anywhere else (in the runner perhaps)? I'm wondering what error you get if you try to queue an `up` for a project that doesn't have a remote data source configured, for example. That is what this was protecting.

Reply:
It did! It got moved directly into the `internal/server/singleprocess/poll_project.go` file. The change is in this PR, there's just a lot in the diff so easy to miss 😅